Pass `BAO_ADDR` to the token helper

ruuda opened the pull request:
Previously the token helper might inherit `BAO_ADDR` from the process, but if the address was specified through an `-address` command-line flag, then the token helper would not know the address, or it would use the wrong one. Fix that by propagating the address everywhere, and then setting `BAO_ADDR` explicitly in the token helper's environment.

Resolves #314.

* [x] One thing I’m not sure about: should we set `VAULT_ADDR` in addition to `BAO_ADDR`? I see there are still many uses of `VAULT_ADDR` in the codebase. ⇒ Yes, set both.

## Target Release

It filled out this by default: 1.14.7

cipherboy commented on command/config/util.go:
```diff
@@ -24,5 +24,12 @@ func DefaultTokenHelper() (token.TokenHelper, error) {
+	// of through an environment variable, we propagate the address to the token
+	// helper through an environment variable. Otherwise the token helper may
+	// read BAO_ADDR and assume a different address than the one we are using.
+	env := []string{"BAO_ADDR=" + vaultAddr}
```
I'd copy from #321 and set `VAULT_ADDR` as well. :-)

cipherboy commented:
\o hello @ruuda -- I think there's some test failures introduced by this PR. If I rebuild this PR on top of my #362, I think in `command/login_test.go` there's several places where `cmd.TokenHelper()` is called without the `vaultAddr` string. 

Sorry for the noisy test failures, I hadn't realized some of my PRs had been merged without passing tests :o

cipherboy commented:
\o hey @ruuda! Now that GA has been released and we're looking at merging more things, do you mind rebasing this and taking a look at the test failures?

If you can't, just let me know and I'm happy to pick it up.

ruuda commented:
Thanks for the reminder! I rebased this and I updated the tests to pass in a dummy address. As far as I can tell none of the tests are exercising the process spawning, so this seems like the easiest way to fix them, though it feels a bit ugly.

cipherboy reviewed (approved):
This looks like a great improvement, thanks @ruuda! 

In the future, it could be cool to expose multi-node token management in the internal helper as well :-) But an improvement for later!
